lib/spack/llnl/util/link_tree.py

```diff
@@ -124,11 +124,61 @@ def unmerge(self, dest_root, **kwargs):
                 if os.path.exists(marker):
                     os.remove(marker)
 
-            elif os.path.exists(dest):
-                if not os.path.islink(dest):
-                    raise ValueError("%s is not a link tree!" % dest)
-                # remove if dest is a hardlink/symlink to src; this will only
-                # be false if two packages are merged into a prefix and have a
-                # conflicting file
-                if filecmp.cmp(src, dest, shallow=True):
-                    os.remove(dest)
+    def merge(self, dest_root, **kwargs):
+        """Link all files in src into dest, creating directories
+           if necessary.
+           If ignore_conflicts is True, do not break when the target exists but
+           rather return a list of files that could not be linked.
+           Note that files blocking directories will still cause an error.
+        """
+        ignore_conflicts = kwargs.get("ignore_conflicts", False)
+
+        ignore = kwargs.get('ignore', lambda x: False)
+        conflict = self.find_conflict(
+            dest_root, ignore=ignore, ignore_file_conflicts=ignore_conflicts)
+        if conflict:
```

Comment:
I'm not sure if it would be more helpful here to take ALL conflicts rather than only the first element of the list

Reply:
I agree but I'm on the fence about whether to add that here: returning a single error is a feature of the API used elsewhere and prior to this (namely in `Package.flatten_dependencies`)